source/docs/essentials/create-a-ui-component.md

```diff
@@ -0,0 +1,307 @@
+---
+id: create-a-ui-component
+title: Create a UI Component
+---
+
+In Front Commerce we split our components in two categories: the **UI**
+components available in the `theme/components` folder, and the business
+components available in the `theme/modules` and `theme/pages` folders.
+
+<blockquote class="info">
+If you feel the need to understand why we went for this organization, feel
+free to refer to [React components structure](#TODO)
+first.
+</blockquote>
+
+In this documentation, you will learn how to build a UI Component. We will do so by
+creating our own. Front-Commerce’s core UI components follow the same principles
+and you could dive into the `node_modules/front-commerce/src/web/theme/components`
+folder to find examples into our source code.
+
+But first, let's define what is an ideal UI Component.
+
+## The ideal UI Component
+
+In Front-Commerce we call UI component any component that is:
+
+- **Reusable in many contexts** If a component is used only once in the whole
+  application, it might be the sign that it does not exist purely for UI
+  purpose. The component most likely needs to be moved to the `theme/modules` folder.
+  That's also the reason why we avoid to give names too close to its business
+  use. For instance, we don't want to have a UI component that would be called
+  `ProductDescription`. It would be better to go for a `Description` that would
+  thus be reusable by a Category.
+- **Focused on abstracting away UI concerns** The goal of UI components is to
+  hide styles or DOM concerns from their parents. It may be hard to achieve
+  sometimes, but it will greatly improve the parent component's readability. For
+  instance, a UI component should not give the opportunity to pass a `className`
+  in its props as it may lead to many styles inconsistencies across the theme.
+
+## How to build a UI Component?
+
+Alright, that's nice in theory, but how does it translate in practice? We'll try
+to get a bit more tangible by creating a UI component needed for adding a
+Reinsurance Banner in a page.
+
+![The reinsurance banner that we will implement](assets/reinsurance.jpg)
+
+### Step 1: Defining the components
```

Comment:
Other pages do not have the "Step x" prefix. I’d suggest removing them here for consistency.

```suggestion
### Defining the components
```